### PR TITLE
PLANNER-1518: Add i18n to React frontend

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
Thanks to @cuijulian for adding the Mandarin translation.

### CODE REVIEW

File: employee-rostering-frontend/translations/cmn/EmployeesPage.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+skillProficiencies: 技能熟练程度
+selectAContract: 选择合同...
+selectSkillProficiencies: 选择技能熟练程度......
```
Comment (ge0ffrey) [inline employee-rostering-frontend/translations/cmn/EmployeesPage.yaml]: Are these ..... intended?

Comment (Christopher-Chianelli) [reply employee-rostering-frontend/translations/cmn/EmployeesPage.yaml]: Typo; Fixed

File: employee-rostering-frontend/translations/en/Alerts.yaml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  title: Error deleting Spot
+  message: The Spot "{{name}}" could not be deleted.
+
+updateSpot:
+  title: Successfully updated Spot
+  message: The Spot with id "{{id}}" was successfully updated.
+
+addTenant:
+  title: Successfully added Tenant
+  message: The Tenant "{{name}}" was succesfully added.
```
Comment (cuijulian) [inline employee-rostering-frontend/translations/en/Alerts.yaml]: Typo

File: employee-rostering-frontend/translations/en/Alerts.yaml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+removeTenant:
+  title: Successfully removed Tenant
+  message: The Tenant "{{name}}" was succesfully deleted.
```
Comment (cuijulian) [inline employee-rostering-frontend/translations/en/Alerts.yaml]: Here as well.

File: employee-rostering-frontend/translations/en/Alerts.yaml (hunk 2 above)
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/translations/en/Alerts.yaml]: Fixed

File: employee-rostering-frontend/translations/en/Alerts.yaml (hunk 3 above)
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/translations/en/Alerts.yaml]: Fixed

File: employee-rostering-frontend/src/ui/pages/shift/__snapshots__/EditShiftModal.test.tsx.snap [hunk 4]
```diff
@@ -122,11 +122,11 @@ exports[`Edit Shift Modal should render correctly when closed 1`] = `
     </InputGroup>
     <InputGroup>
       <Label>
-        Spot
+        Trans(i18nKey=spot)
       </Label>
       <TypeaheadSelectInput
         aria-label="Spot"
```
Comment (MusaTalluzi) [inline employee-rostering-frontend/src/ui/pages/shift/__snapshots__/EditShiftModal.test.tsx.snap]: Shouldn't you also use translation for the aria-label's (applies to many inputs in this file)?

File: employee-rostering-frontend/translations/en/ShiftRosterPage.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+---
+createShift: Create Shift
+noSpots: 
+  <0>There are no Spots</0>
+  <1>
+    The current tenant have no Spots. You need at least one Spot to see the Shift Roster.
+    You can add a Spot in the Spots page.
```
Comment (MusaTalluzi) [inline employee-rostering-frontend/translations/en/ShiftRosterPage.yaml]: For consistency, surround Spots with quotations in "You can add a Spot in the "Spots" page" as in Employee and AvailabilityRoster pages.

File: employee-rostering-frontend/src/ui/__snapshots__/Alerts.test.tsx.snap [hunk 6]
```diff
@@ -63,12 +121,41 @@ exports[`Alerts should render correctly with alerts 1`] = `
...
     onMouseEnter={[Function]}
-    title="alerts.dangerMessage.title"
+    title="Trans(i18nKey=dangerMessage.title)"
     variant="danger"
   >
     <Trans
       components={Array []}
-      i18nKey="alerts.dangerMessage.message"
+      i18nKey="dangerMessage.message"
+      t={
```
[the remark this answers is not in the document]
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/src/ui/__snapshots__/Alerts.test.tsx.snap]: This is a snapshot, not code. Snapshots are used to verify the code is rendering correctly (example: verifying having no spots render the "No spots", and having spots render the schedule).

File: employee-rostering-frontend/src/ui/components/__snapshots__/ServerSideExceptionDialog.test.tsx.snap [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 exports[`ServerSideExceptionDialog should render an ServerSideException Alert correctly 1`] = `
```
[the remark this answers is not in the document]
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/src/ui/components/__snapshots__/ServerSideExceptionDialog.test.tsx.snap]: Wrong place to change it; it should be changed in `src/ui/components/ServerSideExceptionDialog.test.tsx`

File: employee-rostering-frontend/src/ui/pages/shift/__snapshots__/EditShiftModal.test.tsx.snap (hunk 4 above)
Comment (Christopher-Chianelli) [reply employee-rostering-frontend/src/ui/pages/shift/__snapshots__/EditShiftModal.test.tsx.snap]: Probably (although 99% tests use aria-labels to locate elements to click, so this is a major change for a separate PR)

**Pull request merged**